### PR TITLE
some bug fixes and another mixin

### PR DESCRIPTION
**Pull request by coreyauger**
- I have fixed a bug in MuiDropDown.
- Also added some more bindings to MuiMenu and MuiTab
- There is also a start to a mixen that will allow you to "drop anything" onto a react control

### CODE REVIEW

File: core/src/main/scala/chandu0101/scalajs/react/components/mixins/DropAnything.scala [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package chandu0101.scalajs.react.components.mixins
+
+import japgolly.scalajs.react._
+import org.scalajs.dom._
+import org.scalajs.dom.ext.PimpedNodeList
+
+import scala.scalajs.js
+
+/**
+ * Created by chandrasekharkode .
```
Comment (chandu0101) [inline core/src/main/scala/chandu0101/scalajs/react/components/mixins/DropAnything.scala:10]: please fix the comment here ...

File: core/src/main/scala/chandu0101/scalajs/react/components/mixins/DropAnything.scala [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ *
+ * To load resources(javascript,stylesheet) on fly
+ */
+trait DropAnything {
+
+  //var listener: js.Function1[Event, _] = null
```
Comment (chandu0101) [inline core/src/main/scala/chandu0101/scalajs/react/components/mixins/DropAnything.scala:16]: delete all dead/commented code...

File: core/src/main/scala/chandu0101/scalajs/react/components/mixins/DropAnything.scala [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      println("DropAnything componentDidMount")
+      val drop = scope.backend.asInstanceOf[DropAnything]
+      val node = scope.getDOMNode()
+
+      node.addEventListener("drop", drop._onDrop _, false)
+      node.addEventListener("dragenter", drop.onDragEnter _, false)
+      node.addEventListener("dragover", drop.onDragOver _, false)
+
+      //org.scalajs.dom.document.addEventListener("dragover", drop.onDragOver _, false)
+      println(node.attributes)
```
Comment (chandu0101) [inline core/src/main/scala/chandu0101/scalajs/react/components/mixins/DropAnything.scala:54]: remove this println ..

**Comment by oyvindberg** [comment]
I just discovered this too. (edit: this = the Dropdown bug)
There is a better way to fix it- the third parameter _is_ an `Item`, it's just the to/from json dance that goes wrong. 

It can be fixed by wrapping the onChange function with something like this

``` scala
  def wrap(f: (ReactEvent, Int, Item) ⇒ Callback): (ReactEvent, Int, js.Dynamic) ⇒ Callback = {
    (e, idx, obj) ⇒ f(e, idx, Item.fromJson(obj))
  }
```

I can contribute that commit if there`s any need for a released functional dropdown, hehe

**Comment by oyvindberg** [comment]
also @chandu0101 - if there's a need to remove superfluous comments and `println`s (and i would agree to do that ) I guess it's better to just do it in one pass than to make contributors do it, no?

**Comment by chandu0101** [comment]
> also @chandu0101 - if there's a need to remove superfluous comments and printlns (and i would agree to do that ) I guess it's better to just do it in one pass than to make contributors do it, no?

agree , we can do this in one go for now(atleast core module) , but in future PR's it should be done at PR  level itself..., i believe you have an upcoming big PR! would you mind taking care of this ?

**Comment by oyvindberg** [comment]
I have a branch with loads of straightforward updates yes. I'm thinking to delay PRing it until scalajs-react 0.10 is released (or at least has a stable API).  Feel free to change things around until then, because all the diffs are trivial and can be done again (mostly search/replace) .

Let us create an issue for cleaning up comments/formatting/printlns, shall we?

**Comment by chandu0101** [comment]
created an issue https://github.com/chandu0101/scalajs-react-components/issues/25

**Comment by chandu0101** [comment]
@coreyauger :+1:

**Comment by coreyauger** [comment]
Thanks guys.. I will be more careful to remove debug and commented code on future commits.

**Pull request merged**